pex/resolver.py

```diff
@@ -187,6 +188,18 @@ def resolve(self, resolvables, resolvable_set=None):
     processed_packages = {}
     distributions = {}
 
+    def get_possibly_conflicting_wheel_package(cached_wheel_pkg, all_packages):
+      def predicate(x):
+        return (hash(x) != hash(cached_wheel_pkg) and
+               x.name == cached_wheel_pkg.name and
+               x.raw_version == cached_wheel_pkg.raw_version and
+               'file://' in x.url)
```

Comment:
move the hash comparison to the last item so that we have chance to short circulate without the need to run a more expensive hash function.